datasets/_global/gem_energy_ownership/crawler.py

```diff
@@ -5,110 +5,154 @@
+):
+    result = context.lookup("associates", name)
+    if result is None:
+        context.log.warning(f"Potential candidate for associates: {name}")
+    if result and result.associates:
+        for associate_data in result.associates:
+            # Update associates
+            associates_names = associate_data.get("associates_names", [])
+            if associates_names:
+                associates.update(associates_names)
+            # Overwrite names based on which one was matched
+            entity_name = associate_data.get("entity")
+            if entity_name:
+                if name == full_name:
+                    full_name = entity_name
+                else:
+                    name = original_name
+                    original_name = entity_name
+    return full_name, original_name, associates
+
+
 def crawl_company(context: Context, row: Dict[str, str], skipped: Set[str]):
     id = row.pop("entity_id")
-    name = row.pop("entity_name")
+    name = row.pop("name")
+    full_name = row.pop("full_name")
+    aliases = row.pop("name_other")
     # Skip entities
     if name is None or id in SKIP_IDS:
         skipped.add(id)
         return
     original_name = row.pop("name_local")
     reg_country = row.pop("registration_country")
-    lei_code = row.pop("legal_entity_identifier")
+    head_country = row.pop("headquarters_country")
+    lei_code = row.pop("global_legal_entity_identifier_index")
     entity_type = row.pop("entity_type")
+    perm_id = row.pop("permid_refinitiv_permanent_identifier")
+    sp_cap = row.pop("s_p_capital_iq")
+    uk_id = row.pop("uk_companies_house")
+    us_sec_id = row.pop("us_sec_central_index_key")
+    us_eia_id = row.pop("us_eia")
+    br_id = row.pop(
+        "brazil_national_registry_of_legal_entities_federal_revenue_service"
+    )
+    in_id = row.pop(
+        "india_corporate_identification_number_ministry_of_corporate_affairs"
+    )
+    ru_id = row.pop(
+        "russia_uniform_state_register_of_legal_entities_of_russian_federation"
+    )
 
     if entity_type == "legal entity":
         schema = "Company"
+    elif entity_type == "arrangement":
+        schema = "LegalEntity"
     elif entity_type == "state body" or entity_type == "state":
         schema = "PublicBody"
     elif entity_type == "person":
         schema = "Person"
     else:
-        schema = "Company"  # 3 universities end up being companies
+        schema = "Company"
 
     entity = context.make(schema)
     entity.id = context.make_slug(id)
 
+    associates: Set[str] = set()
+    for name in [full_name, original_name]:
+        if name and re.search(NAME_PATTERN, name):
+            full_name, original_name, associates = get_associates(
+                context, name, associates, full_name, original_name
+            )
+
+    if associates:
+        for associate in associates:
+            other = context.make("LegalEntity")
+            other.id = context.make_slug("named", associate)
```

Comment:
let's use make_id instead of make_slug for names, personal identification numbers, etc - things considered possibly personal